hybrid/dispatch/hybrid_dispatch_builder_solver.py

```diff
@@ -501,14 +507,31 @@ def battery_heuristic(self):
             wind_gen = self.power_sources['wind'].dispatch.available_generation
             tot_gen = [wind + gen for wind, gen in zip(wind_gen, tot_gen)]
 
-        grid_limit = self.power_sources['grid'].dispatch.generation_transmission_limit
-
-        if 'one_cycle' in self.options.battery_dispatch:
+        grid_limit = self.power_sources['grid'].dispatch.generation_transmission_limit        
+        if  'baseload' in self.options.battery_dispatch:
+            required_keys = ['limit', 'compliance_factor']
+            if any(key not in self.options.baseload.keys() for key in required_keys):
+                is_missing = [key not in self.options.baseload.keys() for key in required_keys]
+                missing_keys = [missed_key for (missed_key, missing) in zip(required_keys, is_missing) if missing]
+                raise ValueError(type(self).__name__ + " requires the following keys: " + str(missing_keys))
+                
+            # Get difference between baseload demand and power generation and control scenario variables
+            baseload_value = self.options.baseload['limit']
+            baseload_diff =  [(baseload_value - x) for x in tot_gen]
+            self.power_sources['battery'].dispatch.baseload_difference = baseload_diff
+            # Note: grid_limit and goal_power are in MW
```

Comment:
are the units for the baseload dispatch options in MW or KW? in the Grid doc string, it says KW for `'baseload_limit': constant value of firm power you want to provide (kW)`?

Reply:
The user input for firm power is in kW, but the inputs to the dispatch option 'baseload_heuristic' are in MW, which is what this note is trying to say (the code it describes got moved).  I moved the comment down to where the 'baseload_heuristic' options is actually called to clarify this point.